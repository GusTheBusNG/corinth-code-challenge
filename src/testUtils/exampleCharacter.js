export const character = {
  name: 'Anakin Skywalker',
  height: '6',
  mass: 'something',
  hair_color: 'brown',
  birth_year: 'some time long ago',
  species: ['example url species'],
  starships: ['example url starships'],
  films: ['example url fil1s']
}
