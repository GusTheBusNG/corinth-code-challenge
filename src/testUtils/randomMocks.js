export const stubLoading = (message = 'loading') => {
  const random = jest.requireActual('../utils/randomPhrases');
  random.getLoadingQuote = jest.fn(() => message)
}

export const stubError = (message = 'error') => {
  const random = jest.requireActual('../utils/randomPhrases');
  random.getErrorQuote = jest.fn(() => message)
}