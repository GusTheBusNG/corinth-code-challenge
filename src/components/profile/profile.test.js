import React from 'react';
import { create } from 'react-test-renderer';

import Profile from './';
import Films from '../films';
import { stubLoading } from '../../testUtils/randomMocks'
import { character } from '../../testUtils/exampleCharacter';

const createProfile = customProps => 
  create(<Profile {...{...character, ...customProps}} />);


describe('Profile', () => {
  describe('render', () => {
    it('renders correctly', async () => {
      stubLoading()
      const profile = createProfile();
      const films = await profile.root.findByType(Films);

      expect(profile.toJSON()).toMatchSnapshot();
      expect(films.props.films).toEqual(['1']);
    })
  })
})
