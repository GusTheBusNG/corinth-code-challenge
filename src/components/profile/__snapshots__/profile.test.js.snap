// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile render renders correctly 1`] = `
<div
  className="profile"
>
  <h1>
    Hello there 
    Anakin Skywalker
  </h1>
  <p
    className="bio"
  >
    6
    cm | 
    something
    kg | 
    some time long ago
     | 
    brown
  </p>
  <div
    className="info-pill"
  >
    <img
      alt="Species"
      className="icon"
      src="https://res.cloudinary.com/dnglix0tu/image/upload/f_auto/v1/LifeWay-star-wars/Species?_a=ATAMhAA0"
    />
    <p
      className="content"
    >
      loading
    </p>
  </div>
  <div
    className="info-pill"
  >
    <img
      alt="Starship"
      className="icon"
      src="https://res.cloudinary.com/dnglix0tu/image/upload/f_auto/v1/LifeWay-star-wars/Starship?_a=ATAMhAA0"
    />
    <p
      className="content"
    >
      loading
    </p>
  </div>
  <div
    className="films"
  >
    <img
      alt="film"
      className="film-poster"
      src="https://res.cloudinary.com/dnglix0tu/image/upload/f_auto/v1/LifeWay-star-wars/movies/1?_a=ATAMhAA0"
    />
  </div>
</div>
`;
